private/buf/cmd/buf/buf_test.go

```diff
@@ -522,6 +522,7 @@ RPC_REQUEST_RESPONSE_UNIQUE       DEFAULT                  Checks that RPC reque
 RPC_REQUEST_STANDARD_NAME         DEFAULT                  Checks that RPC request type names are RPCNameRequest or ServiceNameRPCNameRequest (configurable).
 RPC_RESPONSE_STANDARD_NAME        DEFAULT                  Checks that RPC response type names are RPCNameResponse or ServiceNameRPCNameResponse (configurable).
 SERVICE_SUFFIX                    DEFAULT                  Checks that services are suffixed with Service (suffix is configurable).
+VALIDATE_CONSTRAINTS_CHECK        DEFAULT                  Checks that fields contain constraints which match the same type.
```

Comment:
Re: previous comment, I think we still need to brainstorm on this a bit. Our other lint checks do not include `_CHECK` in them - I'd like to see some options/a brainstorm with respect to the naming here, can you provide that?

Reply:
1. `VALIDATE_CONSTRAINTS_CONSISTENCY`
1. `VALIDATE_VALIDATION_INTEGRITY`
1. `VALIDATE_FIELD_RULES_CHECK`
1. `VALIDATE_CONFORMANCE_ALL`
1. `VALIDATE_RULES_ALIGNMENT`
1. `VALIDATE_SYNTAX_HARMONY`
1. `VALIDATE_FIELD_CONSISTENCY`
1. `VALIDATE_RULES_COHERENCE`
1. `VALIDATE_VALIDATION_CORRECTNESS`
1. `VALIDATE_STRUCTURE_ALIGNMENT`

Reply:
Can you provide your thinking behind each and how they align with other rules? As an example, (3) continues to use the `CHECK` pattern

Reply:
github has split the thread, linking [here](https://github.com/bufbuild/buf/pull/2377#discussion_r1304635029) for clarity